lib/index.js

```diff
@@ -79,20 +79,30 @@ PostHTML.prototype.process = function (tree, options) {
    * @prop {?Boolean} options.sync - enables sync mode, plugins will run synchronously, throws an error when used with async plugins
    * @prop {?Function} options.parser - use custom parser, replaces default (posthtml-parser)
    * @prop {?Function} options.render - use custom render, replaces default (posthtml-render)
-   * @prop {?Boolean} options.skipParse - disable parsing
    */
   options = options || {}
 
+  if (typeof tree !== 'string' || typeof tree !== 'object') {
+    throw new TypeError(
+      'Input must to be a String (HTML) or Object (PostHTML AST)'
+    )
+  }
+
+  tree = typeof tree === 'object' ? tree : parser(tree)
+
+  tree.parser = parser
```

Comment:
and plugins shoud support all parsers? ) bad idea
why not use gulp/builders/streams?